### PR TITLE
UI clean up

### PR DESCRIPTION
**Pull request by JayGray**
Main goal of this pull request is to clean up the UI code.

First of all I removed all obsolete angular code and build environment of the former UI.
After this I "improved" the development workflow, now only one command is needed to
fire up the UI in dev mode. I did not care about the production build setup for now.
Furthermore some small refactoring of react code. (Did not care about code styling or other thing yet)
Splitted VIEWS, COMPONENTS and ROUTES into separate files/folders, added a main
entry point of the app.

This PR as once looks huge but going through each commit will lighten up
many of the things I have done.

I tried to be precise as possible in the commit messages.

Happy reviewing ;)

### CODE REVIEW

**Comment by JayGray** [comment]
@mschaaf, @thschlender please have a look and share your thoughts.

File: ui/app/src/components/Co2Bar.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  constructor() {
+    super();
+    this.state = {
+      co2: 0.0,
+      treesCount: 0
+    };
+  }
+
+  componentDidMount() {
+    var that = this;
```
Comment (thschlender) [inline ui/app/src/components/Co2Bar.js]: Ist es nicht besser/sicherer var mit let zu ersetzen? Momentan ist es OK, aber wenn die Struktur in der Methode komplexer wird, könnte man that ungewollt überschreiben: https://www.youtube.com/watch?v=1sHq04erG9o

Comment (JayGray) [inline ui/app/src/components/Co2Bar.js]: Auf alle Fälle. Doch wie bereits erwähnt ist es erstmal nur ein clean up.

File: ui/package.json [hunk 2]
```diff
@@ -17,31 +17,14 @@
...
-    "karma-mocha-reporter": "^1.0.2",
-    "karma-ng-html2js-preprocessor": "~0.1.0",
-    "karma-ng-scenario": "~0.1.0",
-    "karma-phantomjs-launcher": "~0.1.4",
-    "karma-script-launcher": "~0.1.0",
-    "karma-sinon-chai": "^1.0.0",
-    "mocha": "^2.2.5",
-    "open": "~0.0.5",
-    "supertest": "~0.11.0",
+    "html-webpack-plugin": "^2.17.0",
```
Comment (mschaaf) [inline ui/package.json]: For what is it?

File: ui/package.json [hunk 3]
```diff
@@ -17,31 +17,14 @@
...
     "webpack": "^1.13.0",
     "webpack-dev-server": "^1.14.1"
   },
   "engines": {
     "node": ">=5.5.0"
   },
   "scripts": {
-    "update-webdriver": "node node_modules/grunt-protractor-runner/node_modules/protractor/bin/webdriver-manager update"
+    "dev": "NODE_ENV=development webpack-dev-server --hot --inline --progress"
```
Comment (mschaaf) [inline ui/package.json]: So this removes the need for http-server plugin?

File: ui/webpack.config.js [hunk 4]
```diff
@@ -1,22 +1,64 @@
...
+        'modulesDirectories': ['node_modules']
+    };
+
+    // Define entry points
+    config.entry = {
+        'app': __dirname + '/app/src/app.js'
+    };
+
+    config.output = {
+        'path': __dirname + '/app/dist/',
```
Comment (mschaaf) [inline ui/webpack.config.js]: I don't like the idea to have the dist folder inside the src folder and the folder we serve the dev environment from. This should be on the same level as app/. I would like to create a build folder in ui and create the dist inside that. Something like: /build/dist/
But maybe I don't understand the proposed structure and you need to explain it more.

**Comment by mschaaf** [comment]
Whats the command now for building the src less dist? Before we could run only ```webpack``` and the dist was ready for deployment in the client-react folder. Now everything src code and static stuff and dist mixed under the app folder. Thats my only but big issue with the changes.

**Comment by mschaaf** [comment]
Another thing I recognized in your branch is that now the bundle.js is 2.37 MB in size and before it was only 937Kb. Why is it now so big?

File: ui/package.json (hunk 3 above)
Comment (JayGray) [inline ui/package.json]: It is removed.

File: ui/webpack.config.js (hunk 4 above)
Comment (JayGray) [inline ui/webpack.config.js]: As I said in the description.
> I did not care about the production build setup for now.

File: ui/package.json (hunk 2 above)
Comment (JayGray) [inline ui/package.json]: Have a look at the description: cdf9267952168ddfcb1de84d00d956903affeb8d

**Comment by JayGray** [comment]
> Another thing I recognized in your branch is that now the bundle.js is 2.37 MB in size and before it was only 937Kb. Why is it now so big?

These is because of the source maps. Before there where no source maps at all. Just a link to the source.

**Comment by JayGray** [comment]
@mschaaf,
thanks for the feedback.

Due to your recommendations I splitted the source and the distribution code into folder (src, dist).
Also add the build task. Available as `npm run build`.
The build task will output the `index.html` with correct paths to th generated js files.
The js files are located in `dist/js`. Now there are 2. One for all the vendor code the other just our
code. Both have a hash in the name this is good for browser caching. After an update if the our changes
but the vendor code is the same the hash should not change and browser can fetch from cache.

Also added some additional optimization for the production code. Deduplication, minification and external source maps. 
Below a table of the output with sizes.
````
25K May 27 14:24 app.9ec79dd415fce3f176b4.js
94K May 27 14:24 app.9ec79dd415fce3f176b4.js.map
214K May 27 14:24 vendor.9ec79dd415fce3f176b4.js
2.0M May 27 14:24 vendor.9ec79dd415fce3f176b4.js.map
```

**Comment by mschaaf** [comment]
Thanks for the latest changes. I will try it as fas as poaaible.

**Comment by JayGray** [comment]
@mschaaf,
I recently rebased with current master, so you might have to delete your local copy and checkout again.

**Comment by mschaaf** [comment]
The explore page is not merged correctly after rebase with master. Push force can be evil ;-)

**Comment by JayGray** [comment]
@mschaaf,

mmmh, und nu?
I will have a look.

**Comment by JayGray** [comment]
BTW. I don't  think it's an issue with `-f` because I rebased before, had one conflict with the deleted `components/Main.js` but everything else was clean. So if I did it w/o `-f`, I guess the same would happen.

**Comment by mschaaf** [comment]
Can you please review my latest changes. I am short before merging. Thanks.

**Comment by JayGray** [comment]
@mschaaf,
[IMveryOO](https://translate.google.de/#en/de/In%20my%20very%20own%20opinion) I would like to see the projects page more as a view rather than a component. BUT please go ahead merge looks good so far. We can split it afterwards.

**Comment by mschaaf** [comment]
Its not a projects page its a page for exploring and part of it is a list fo projects. But at the end the ipat guys have to decide what they really want.

**Comment by mschaaf** [comment]
Thanks its now in master.

**Comment by JayGray** [comment]
YAY: 🎉

**Pull request merged**